xdsl/ir/affine/affine_set.py

```diff
@@ -0,0 +1,48 @@
+from dataclasses import dataclass
+from enum import Enum
+
+from xdsl.ir.affine import AffineExpr
+
+
+class AffineConstraintKind(Enum):
+    ge = ">="
+    le = "<="
+    eq = "=="
+
+
+@dataclass
+class AffineConstraintExpr:
+    kind: AffineConstraintKind
+    lhs: AffineExpr
+    rhs: AffineExpr
+
+    def __post_init__(self):
+        self.lhs = self.lhs - self.rhs
+        self.rhs = AffineExpr.constant(0)
```

Comment:
```suggestion
    def __init__(self, kind: AffineConstraintKind, lhs: AffineExpr, rhs: AffineExpr, *, canonicalize: bool = True):
        if canonicalize:
            lhs = self.lhs - self.rhs
            rhs = AffineExpr.constant(0)
        self.kind = kind
        self.lhs = lhs
        self.rhs = rhs
```

Can you please make it possible to opt out of this canonicalization?